### ISSUE TITLE
[bug]: failed to update ride-the-lightning after restoring Embassy OS from backup

### ISSUE THREAD

**Issue by swissrouting** [Start9Labs/embassy-os#1550]
### Prerequisites

- [X] I have searched for [existing issues](https://github.com/start9labs/embassy-os/issues) that already report this problem, without success.

### EmbassyOS Version

0.3.1 (built from GitHub master)

### Device

Laptop/Desktop

### Device OS

Windows

### Device OS Version

10

### Browser

Chrome

### Browser Version

102.0.5005.115

### Current Behavior

After clicking "Update" on RTL app in marketplace, Embassy goes through these steps:

1. Downloading (reached 100%)
2. Validating (reached 100%)
3. Unpacking (reached 100%)

Then RTL remains stuck showing status `Finalizing install. This could take a minute...`

### Expected Behavior

RTL should update successfully and the message `Finalizing install. This could take a minute...` should disappear eventually.

### Steps to Reproduce

Initial state:
- Embassy device running eOS 0.3.0.3 (latest released version)
- Bitcoin Core, Bitcoin Proxy, and Ride The Lightning (RTL) apps installed
- Device backed up to LAN drive

Steps to reproduce:
1. Build eOS from master (at commit b0068a333bc684ec989a51c92c2004c2e57edd6b)
2. Install new eOS build on SD card and insert into Embassy
3. Restore from LAN (overwrite contents of attached SSD, keepsame master password)
4. Switch to beta marketplace
5. Attempt to update RTL to latest version

### Anything else?

Log collected over SSH using: `$ sudo journalctl -u embassyd -efa`

```
Jun 16 22:37:22 embassy-6e79b9cc embassyd[1352]: Jun 16 22:37:22.375  INFO download_install_s9pk{marketplace_url=Some(Url { scheme: "https", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("beta-registry-0-3.start9labs.com")), port: None, path: "/", query: None, fragment: None }) progress=InstallProgress { size: Some(231697634), downloaded: 0, download_complete: false, validated: 0, validation_complete: false, unpacked: 0, unpack_complete: false }}:install_s9pk{pkg_id=PackageId(Id("ride-the-lightning")) version=Version { version: Version(0, 12, 3, 1), string: "0.12.3.1" } marketplace_url=Some(Url { scheme: "https", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("beta-registry-0-3.start9labs.com")), port: None, path: "/", query: None, fragment: None }) progress=InstallProgress { size: Some(231697634), downloaded: 231697634, download_complete: true, validated: 231697634, validation_complete: false, unpacked: 0, unpack_complete: false }}: embassy::install: Install ride-the-lightning@0.12.3.1: Creating volumes
Jun 16 22:37:22 embassy-6e79b9cc embassyd[1352]: Jun 16 22:37:22.376  INFO download_install_s9pk{marketplace_url=Some(Url { scheme: "https", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("beta-registry-0-3.start9labs.com")), port: None, path: "/", query: None, fragment: None }) progress=InstallProgress { size: Some(231697634), downloaded: 0, download_complete: false, validated: 0, validation_complete: false, unpacked: 0, unpack_complete: false }}:install_s9pk{pkg_id=PackageId(Id("ride-the-lightning")) version=Version { version: Version(0, 12, 3, 1), string: "0.12.3.1" } marketplace_url=Some(Url { scheme: "https", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("beta-registry-0-3.start9labs.com")), port: None, path: "/", query: None, fragment: None }) progress=InstallProgress { size: Some(231697634), downloaded: 231697634, download_complete: true, validated: 231697634, validation_complete: false, unpacked: 0, unpack_complete: false }}: embassy::install: Install ride-the-lightning@0.12.3.1: Created volumes
Jun 16 22:37:22 embassy-6e79b9cc embassyd[1352]: Jun 16 22:37:22.376  INFO download_install_s9pk{marketplace_url=Some(Url { scheme: "https", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("beta-registry-0-3.start9labs.com")), port: None, path: "/", query: None, fragment: None }) progress=InstallProgress { size: Some(231697634), downloaded: 0, download_complete: false, validated: 0, validation_complete: false, unpacked: 0, unpack_complete: false }}:install_s9pk{pkg_id=PackageId(Id("ride-the-lightning")) version=Version { version: Version(0, 12, 3, 1), string: "0.12.3.1" } marketplace_url=Some(Url { scheme: "https", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("beta-registry-0-3.start9labs.com")), port: None, path: "/", query: None, fragment: None }) progress=InstallProgress { size: Some(231697634), downloaded: 231697634, download_complete: true, validated: 231697634, validation_complete: false, unpacked: 0, unpack_complete: false }}: embassy::install: Install ride-the-lightning@0.12.3.1: Installing interfaces
Jun 16 22:37:22 embassy-6e79b9cc embassyd[1352]: Jun 16 22:37:22.376  INFO download_install_s9pk{marketplace_url=Some(Url { scheme: "https", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("beta-registry-0-3.start9labs.com")), port: None, path: "/", query: None, fragment: None }) progress=InstallProgress { size: Some(231697634), downloaded: 0, download_complete: false, validated: 0, validation_complete: false, unpacked: 0, unpack_complete: false }}:install_s9pk{pkg_id=PackageId(Id("ride-the-lightning")) version=Version { version: Version(0, 12, 3, 1), string: "0.12.3.1" } marketplace_url=Some(Url { scheme: "https", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("beta-registry-0-3.start9labs.com")), port: None, path: "/", query: None, fragment: None }) progress=InstallProgress { size: Some(231697634), downloaded: 231697634, download_complete: true, validated: 231697634, validation_complete: false, unpacked: 0, unpack_complete: false }}: embassy::install: Install ride-the-lightning@0.12.3.1: Installed interfaces
Jun 16 22:37:22 embassy-6e79b9cc embassyd[1352]: Jun 16 22:37:22.377  INFO download_install_s9pk{marketplace_url=Some(Url { scheme: "https", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("beta-registry-0-3.start9labs.com")), port: None, path: "/", query: None, fragment: None }) progress=InstallProgress { size: Some(231697634), downloaded: 0, download_complete: false, validated: 0, validation_complete: false, unpacked: 0, unpack_complete: false }}:install_s9pk{pkg_id=PackageId(Id("ride-the-lightning")) version=Version { version: Version(0, 12, 3, 1), string: "0.12.3.1" } marketplace_url=Some(Url { scheme: "https", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("beta-registry-0-3.start9labs.com")), port: None, path: "/", query: None, fragment: None }) progress=InstallProgress { size: Some(231697634), downloaded: 231697634, download_complete: true, validated: 231697634, validation_complete: false, unpacked: 0, unpack_complete: false }}: embassy::install: Install ride-the-lightning@0.12.3.1: Creating manager
Jun 16 22:37:22 embassy-6e79b9cc embassyd[1352]: Jun 16 22:37:22.378  INFO download_install_s9pk{marketplace_url=Some(Url { scheme: "https", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("beta-registry-0-3.start9labs.com")), port: None, path: "/", query: None, fragment: None }) progress=InstallProgress { size: Some(231697634), downloaded: 0, download_complete: false, validated: 0, validation_complete: false, unpacked: 0, unpack_complete: false }}:install_s9pk{pkg_id=PackageId(Id("ride-the-lightning")) version=Version { version: Version(0, 12, 3, 1), string: "0.12.3.1" } marketplace_url=Some(Url { scheme: "https", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("beta-registry-0-3.start9labs.com")), port: None, path: "/", query: None, fragment: None }) progress=InstallProgress { size: Some(231697634), downloaded: 231697634, download_complete: true, validated: 231697634, validation_complete: false, unpacked: 0, unpack_complete: false }}: embassy::install: Install ride-the-lightning@0.12.3.1: Created manager
Jun 16 22:37:26 embassy-6e79b9cc embassyd[1352]: Jun 16 22:37:26.020  WARN embassy::procedure::docker: Failed to deserialize stdout from JSON: Deserialization Error: EOF while parsing a value at line 1 column 0, falling back to UTF-8 string.
Jun 16 22:37:51 embassy-6e79b9cc embassyd[1352]: Jun 16 22:37:51.933 ERROR embassy::install: Install of ride-the-lightning@0.12.3.1 Failed: Migration Failed: Error: stat /root/start9/config.yaml: no such file or directory
Jun 16 22:37:51 embassy-6e79b9cc embassyd[1352]: Jun 16 22:37:51.933 DEBUG embassy::install: Error { source:
Jun 16 22:37:51 embassy-6e79b9cc embassyd[1352]:    0: Error: stat /root/start9/config.yaml: no such file or directory
Jun 16 22:37:51 embassy-6e79b9cc embassyd[1352]: Location:
Jun 16 22:37:51 embassy-6e79b9cc embassyd[1352]:    /home/rust/src/backend/src/migration.rs:77
Jun 16 22:37:51 embassy-6e79b9cc embassyd[1352]:   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ SPANTRACE ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
Jun 16 22:37:51 embassy-6e79b9cc embassyd[1352]:    0: embassy::install::install_s9pk with pkg_id=PackageId(Id("ride-the-lightning")) version=Version { version: Version(0, 12, 3, 1), string: "0.12.3.1" } marketplace_url=Some(Url { scheme: "https", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("beta-registry-0-3.start9labs.com")), port: None, path: "/", query: None, fragment: None }) progress=InstallProgress { size: Some(231697634), downloaded: 231697634, download_complete: true, validated: 231697634, validation_complete: false, unpacked: 0, unpack_complete: false }
Jun 16 22:37:51 embassy-6e79b9cc embassyd[1352]:       at src/install/mod.rs:884
Jun 16 22:37:51 embassy-6e79b9cc embassyd[1352]:    1: embassy::install::download_install_s9pk with marketplace_url=Some(Url { scheme: "https", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("beta-registry-0-3.start9labs.com")), port: None, path: "/", query: None, fragment: None }) progress=InstallProgress { size: Some(231697634), downloaded: 0, download_complete: false, validated: 0, validation_complete: false, unpacked: 0, unpack_complete: false }
Jun 16 22:37:51 embassy-6e79b9cc embassyd[1352]:       at src/install/mod.rs:722
Jun 16 22:37:51 embassy-6e79b9cc embassyd[1352]: Backtrace omitted.
Jun 16 22:37:51 embassy-6e79b9cc embassyd[1352]: Run with RUST_BACKTRACE=1 environment variable to display it.
Jun 16 22:37:51 embassy-6e79b9cc embassyd[1352]: Run with RUST_BACKTRACE=full to include source snippets., kind: MigrationFailed, revision: None }
```

I confirmed that the file does not exist, and in fact the directory `/embassy-data/package-data/volumes/ride-the-lightning/data/main/` is entirely empty.

**Comment by chrisguida** [comment]
Do you have any earlier logs that show the backup restore happening? Seems it failed, hopefully there are error messages explaining why

**Comment by swissrouting** [comment]
I looked through all earlier logs using `sudo journalctl -u embassyd`. This is what I see at the very beginning:

```
Jun 16 21:55:11 embassy systemd[1]: Started Embassy Daemon.
Jun 16 21:55:12 embassy-6e79b9cc embassyd[6277]: Jun 16 21:55:12.747  INFO inner_main{cfg_path=None}:init{disk_guid="EMBASSY_OS2AQ_REDACTED_LAQ"}: embassy::context::rpc: Loaded Config
Jun 16 21:55:12 embassy-6e79b9cc embassyd[6277]: Jun 16 21:55:12.875  INFO inner_main{cfg_path=None}:init{disk_guid="EMBASSY_OS2AQ_REDACTED_LAQ"}: embassy::context::rpc: Opened Sqlite DB
Jun 16 21:55:13 embassy-6e79b9cc embassyd[6277]: Jun 16 21:55:13.699  INFO inner_main{cfg_path=None}:init{disk_guid="EMBASSY_OS2AQ_REDACTED_LAQ"}: embassy::context::rpc: Opened PatchDB
Jun 16 21:55:13 embassy-6e79b9cc embassyd[6277]: Jun 16 21:55:13.710  INFO inner_main{cfg_path=None}:init{disk_guid="EMBASSY_OS2AQ_REDACTED_LAQ"}: embassy::context::rpc: Connected to Docker
Jun 16 21:55:13 embassy-6e79b9cc embassyd[6277]: Jun 16 21:55:13.816  INFO inner_main{cfg_path=None}:init{disk_guid="EMBASSY_OS2AQ_REDACTED_LAQ"}:init{embassyd_addr=127.0.0.1:80 embassyd_tor_key=TorSecretKey(****) tor_control=127.0.0.1:9051 import_root_ca=None}:init{embassyd_addr=127.0.0.1:80 embassyd_tor_key=TorSecretKey(****) tor_control=127.0.0.1:9051}:add_embassyd_onion: embassy::net::tor: Registering Main Tor Service: REDACTED.onion
Jun 16 21:55:14 embassy-6e79b9cc embassyd[6277]: Jun 16 21:55:14.224  INFO inner_main{cfg_path=None}:init{disk_guid="EMBASSY_OS2AQ_REDACTED_LAQ"}:init{embassyd_addr=127.0.0.1:80 embassyd_tor_key=TorSecretKey(****) tor_control=127.0.0.1:9051 import_root_ca=None}:init{embassyd_addr=127.0.0.1:80 embassyd_tor_key=TorSecretKey(****) tor_control=127.0.0.1:9051}:add_embassyd_onion: embassy::net::tor: Registered Main Tor Service: REDACTED.onion
Jun 16 21:55:14 embassy-6e79b9cc embassyd[6277]: Jun 16 21:55:14.224 DEBUG inner_main{cfg_path=None}:init{disk_guid="EMBASSY_OS2AQ_REDACTED_LAQ"}:init{embassyd_addr=127.0.0.1:80 embassyd_tor_key=TorSecretKey(****) tor_control=127.0.0.1:9051 import_root_ca=None}: embassy::net::mdns: Initializing mDNS controller
Jun 16 21:55:14 embassy-6e79b9cc embassyd[6277]: Jun 16 21:55:14.236  WARN inner_main{cfg_path=None}:init{disk_guid="EMBASSY_OS2AQ_REDACTED_LAQ"}:init{embassyd_addr=127.0.0.1:80 embassyd_tor_key=TorSecretKey(****) tor_control=127.0.0.1:9051 import_root_ca=None}: embassy::net::mdns: AvahiCallback: ClientState = AVAHI_CLIENT_S_RUNNING
Jun 16 21:55:14 embassy-6e79b9cc embassyd[6277]: Jun 16 21:55:14.240  WARN inner_main{cfg_path=None}:init{disk_guid="EMBASSY_OS2AQ_REDACTED_LAQ"}:init{embassyd_addr=127.0.0.1:80 embassyd_tor_key=TorSecretKey(****) tor_control=127.0.0.1:9051 import_root_ca=None}: embassy::net::mdns: AvahiCallback: EntryGroupState = AVAHI_ENTRY_GROUP_UNCOMMITED
Jun 16 21:55:14 embassy-6e79b9cc embassyd[6277]: Jun 16 21:55:14.242 DEBUG inner_main{cfg_path=None}:init{disk_guid="EMBASSY_OS2AQ_REDACTED_LAQ"}:init{embassyd_addr=127.0.0.1:80 embassyd_tor_key=TorSecretKey(****) tor_control=127.0.0.1:9051 import_root_ca=None}: embassy::net::mdns: Loading services for mDNS
Jun 16 21:55:14 embassy-6e79b9cc embassyd[6277]: Jun 16 21:55:14.244  INFO inner_main{cfg_path=None}:init{disk_guid="EMBASSY_OS2AQ_REDACTED_LAQ"}:init{embassyd_addr=127.0.0.1:80 embassyd_tor_key=TorSecretKey(****) tor_control=127.0.0.1:9051 import_root_ca=None}: embassy::net::mdns: Published "embassy-6e79b9cc.local"
Jun 16 21:55:14 embassy-6e79b9cc embassyd[6277]: Jun 16 21:55:14.253  INFO inner_main{cfg_path=None}:init{disk_guid="EMBASSY_OS2AQ_REDACTED_LAQ"}: embassy::context::rpc: Initialized Net Controller
Jun 16 21:55:14 embassy-6e79b9cc embassyd[6277]: Jun 16 21:55:14.253  INFO inner_main{cfg_path=None}:init{disk_guid="EMBASSY_OS2AQ_REDACTED_LAQ"}: embassy::context::rpc: Initialized Notification Manager
Jun 16 21:55:14 embassy-6e79b9cc embassyd[6277]: Jun 16 21:55:14.329  INFO inner_main{cfg_path=None}:init{disk_guid="EMBASSY_OS2AQ_REDACTED_LAQ"}: embassy::context::rpc: Cleaned up transient states
Jun 16 21:55:14 embassy-6e79b9cc embassyd[6277]: Jun 16 21:55:14.361  INFO inner_main{cfg_path=None}:init{disk_guid="EMBASSY_OS2AQ_REDACTED_LAQ"}: embassy::context::rpc: Initialized Package Managers
Jun 16 21:56:39 embassy-6e79b9cc embassyd[6277]: Jun 16 21:56:39.012 DEBUG embassyd: Request to /ws/db
Jun 16 21:58:06 embassy-6e79b9cc embassyd[6277]: Jun 16 21:58:06.897 DEBUG unmount: embassy::disk::mount::util: Unmounting /media/embassy-os/tmp/4WBECWRQ3EBVEDOVBUVEFVYTQGFCRE32QGATEIVQLLCLZBCSST5A.
Jun 16 21:59:41 embassy-6e79b9cc mount.ecryptfs[6370]: Error initializing key module [/usr/lib/aarch64-linux-gnu/ecryptfs/libecryptfs_key_mod_gpg.so]; rc = [-22]
Jun 16 21:59:41 embassy-6e79b9cc embassyd[6370]: Filename Encryption Key (FNEK) Signature [f14f0b5f1b2512e6]: Attempting to mount with the following options:
Jun 16 21:59:41 embassy-6e79b9cc embassyd[6370]:   ecryptfs_unlink_sigs
Jun 16 21:59:41 embassy-6e79b9cc embassyd[6370]:   ecryptfs_fnek_sig=f14f0b5f1b2512e6
Jun 16 21:59:41 embassy-6e79b9cc embassyd[6370]:   ecryptfs_key_bytes=32
Jun 16 21:59:41 embassy-6e79b9cc embassyd[6370]:   ecryptfs_cipher=aes
Jun 16 21:59:41 embassy-6e79b9cc embassyd[6370]:   ecryptfs_sig=f14f0b5f1b2512e6
Jun 16 21:59:41 embassy-6e79b9cc embassyd[6370]: Mounted eCryptfs
Jun 16 21:59:42 embassy-6e79b9cc embassyd[6277]: Jun 16 21:59:42.004  INFO perform_backup:mount_package_backup{id=PackageId(Id("bitcoind"))}:bind{read_only=false}: embassy::disk::mount::util: Binding /media/embassy-os/tmp/35SIONHD462AVVDVLRYJC5NUPD3RKXHEOIAFB6QF4GGBTUCDXGMQ/bitcoind to /media/embassy-os/backups/bitcoind
Jun 16 22:00:15 embassy-6e79b9cc embassyd[6277]: Jun 16 22:00:15.276 DEBUG embassy::net::tor: Attempting to self-check tor address
Jun 16 22:00:18 embassy-6e79b9cc embassyd[6277]: Jun 16 22:00:18.622 DEBUG perform_backup:unmount: embassy::disk::mount::util: Unmounting /media/embassy-os/backups/bitcoind.
Jun 16 22:00:18 embassy-6e79b9cc embassyd[6277]: Jun 16 22:00:18.655  INFO perform_backup:mount_package_backup{id=PackageId(Id("btc-rpc-proxy"))}:bind{read_only=false}: embassy::disk::mount::util: Binding /media/embassy-os/tmp/35SIONHD462AVVDVLRYJC5NUPD3RKXHEOIAFB6QF4GGBTUCDXGMQ/btc-rpc-proxy to /media/embassy-os/backups/btc-rpc-proxy
Jun 16 22:00:18 embassy-6e79b9cc embassyd[6277]: Jun 16 22:00:18.939 DEBUG embassy::net::tor: Successfully verified main tor address liveness at REDACTED.onion
Jun 16 22:00:24 embassy-6e79b9cc embassyd[6277]: Jun 16 22:00:24.489 DEBUG perform_backup:unmount: embassy::disk::mount::util: Unmounting /media/embassy-os/backups/btc-rpc-proxy.
Jun 16 22:00:24 embassy-6e79b9cc umount.ecryptfs[6682]: Failed to find key with sig [f14f0b5f1b2512e6]: Required key not available
Jun 16 22:00:24 embassy-6e79b9cc embassyd[6277]: Jun 16 22:00:24.523  INFO perform_backup:mount_package_backup{id=PackageId(Id("c-lightning"))}:bind{read_only=false}: embassy::disk::mount::util: Binding /media/embassy-os/tmp/35SIONHD462AVVDVLRYJC5NUPD3RKXHEOIAFB6QF4GGBTUCDXGMQ/c-lightning to /media/embassy-os/backups/c-lightning
Jun 16 22:01:26 embassy-6e79b9cc embassyd[6277]: Jun 16 22:01:26.709 DEBUG perform_backup:unmount: embassy::disk::mount::util: Unmounting /media/embassy-os/backups/c-lightning.
Jun 16 22:01:26 embassy-6e79b9cc umount.ecryptfs[6835]: Failed to find key with sig [f14f0b5f1b2512e6]: Required key not available
Jun 16 22:01:26 embassy-6e79b9cc embassyd[6277]: Jun 16 22:01:26.753  INFO perform_backup:mount_package_backup{id=PackageId(Id("filebrowser"))}:bind{read_only=false}: embassy::disk::mount::util: Binding /media/embassy-os/tmp/35SIONHD462AVVDVLRYJC5NUPD3RKXHEOIAFB6QF4GGBTUCDXGMQ/filebrowser to /media/embassy-os/backups/filebrowser
Jun 16 22:01:33 embassy-6e79b9cc embassyd[6277]: Jun 16 22:01:33.812 DEBUG perform_backup:unmount: embassy::disk::mount::util: Unmounting /media/embassy-os/backups/filebrowser.
Jun 16 22:01:33 embassy-6e79b9cc umount.ecryptfs[6983]: Failed to find key with sig [f14f0b5f1b2512e6]: Required key not available
Jun 16 22:01:33 embassy-6e79b9cc embassyd[6277]: Jun 16 22:01:33.846  INFO perform_backup:mount_package_backup{id=PackageId(Id("mempool"))}:bind{read_only=false}: embassy::disk::mount::util: Binding /media/embassy-os/tmp/35SIONHD462AVVDVLRYJC5NUPD3RKXHEOIAFB6QF4GGBTUCDXGMQ/mempool to /media/embassy-os/backups/mempool
Jun 16 22:02:51 embassy-6e79b9cc embassyd[6277]: Jun 16 22:02:51.371 DEBUG perform_backup:unmount: embassy::disk::mount::util: Unmounting /media/embassy-os/backups/mempool.
Jun 16 22:02:51 embassy-6e79b9cc umount.ecryptfs[7137]: Failed to find key with sig [f14f0b5f1b2512e6]: Required key not available
Jun 16 22:02:51 embassy-6e79b9cc embassyd[6277]: Jun 16 22:02:51.406  INFO perform_backup:mount_package_backup{id=PackageId(Id("ride-the-lightning"))}:bind{read_only=false}: embassy::disk::mount::util: Binding /media/embassy-os/tmp/35SIONHD462AVVDVLRYJC5NUPD3RKXHEOIAFB6QF4GGBTUCDXGMQ/ride-the-lightning to /media/embassy-os/backups/ride-the-lightning
Jun 16 22:03:36 embassy-6e79b9cc embassyd[6277]: Jun 16 22:03:36.242 DEBUG perform_backup:unmount: embassy::disk::mount::util: Unmounting /media/embassy-os/backups/ride-the-lightning.
Jun 16 22:03:36 embassy-6e79b9cc umount.ecryptfs[7285]: Failed to find key with sig [f14f0b5f1b2512e6]: Required key not available
Jun 16 22:03:36 embassy-6e79b9cc embassyd[6277]: Jun 16 22:03:36.267  INFO perform_backup:mount_package_backup{id=PackageId(Id("synapse"))}:bind{read_only=false}: embassy::disk::mount::util: Binding /media/embassy-os/tmp/35SIONHD462AVVDVLRYJC5NUPD3RKXHEOIAFB6QF4GGBTUCDXGMQ/synapse to /media/embassy-os/backups/synapse
Jun 16 22:04:15 embassy-6e79b9cc embassyd[6277]: Jun 16 22:04:15.048 DEBUG perform_backup:unmount: embassy::disk::mount::util: Unmounting /media/embassy-os/backups/synapse.
Jun 16 22:04:15 embassy-6e79b9cc umount.ecryptfs[7445]: Failed to find key with sig [f14f0b5f1b2512e6]: Required key not available
Jun 16 22:04:15 embassy-6e79b9cc embassyd[6277]: Jun 16 22:04:15.106  INFO perform_backup:mount_package_backup{id=PackageId(Id("vaultwarden"))}:bind{read_only=false}: embassy::disk::mount::util: Binding /media/embassy-os/tmp/35SIONHD462AVVDVLRYJC5NUPD3RKXHEOIAFB6QF4GGBTUCDXGMQ/vaultwarden to /media/embassy-os/backups/vaultwarden
Jun 16 22:04:40 embassy-6e79b9cc embassyd[6277]: Jun 16 22:04:40.428 DEBUG perform_backup:unmount: embassy::disk::mount::util: Unmounting /media/embassy-os/backups/vaultwarden.
Jun 16 22:04:40 embassy-6e79b9cc umount.ecryptfs[7595]: Failed to find key with sig [f14f0b5f1b2512e6]: Required key not available
Jun 16 22:04:40 embassy-6e79b9cc embassyd[6277]: Jun 16 22:04:40.472 DEBUG perform_backup:save_and_unmount:unmount:unmount: embassy::disk::mount::util: Unmounting /media/embassy-os/tmp/35SIONHD462AVVDVLRYJC5NUPD3RKXHEOIAFB6QF4GGBTUCDXGMQ.
Jun 16 22:04:40 embassy-6e79b9cc umount.ecryptfs[7597]: Failed to find key with sig [f14f0b5f1b2512e6]: Required key not available
Jun 16 22:04:40 embassy-6e79b9cc embassyd[6277]: Jun 16 22:04:40.822 DEBUG perform_backup:save_and_unmount:unmount:unmount: embassy::disk::mount::util: Unmounting /media/embassy-os/tmp/4WBECWRQ3EBVEDOVBUVEFVYTQGFCRE32QGATEIVQLLCLZBCSST5A.
```

Looks like this is iterating through all the services that I had installed previously and were included in the backup. This includes bitcoin and ride-the-lightning. But it's failing somewhere with `Required key not available`.

Does this provide enough info to make a guess at why the restore could have failed? Or are there some other diagnostic commands I could try?

Note: this is a dev machine with nothing important, I am happy to throw it away and start fresh. I will only keep it around as long as it's helpful to solve this bug.

**Comment by dr-bonez** [comment]
the `Required key not available` can be ignored. It is a result of how we use `ecryptfs` without storing encryption keys in the kernel keyring. 

there is clearly a bug in the migration for RTL, however, the OS *should* have issued a notification with an error message, and the service should have disappeared. There was likely an issue cleaning up the service. do you have any errors *later* in the logs?

**Comment by MattDHill** [comment]
Should be fixed in 0.3.1.1

**Comment by chrisguida** [comment]
What was the problem, and where was it fixed?
